test/extensions/filters/http/oauth2/oauth_test.cc

```diff
@@ -263,6 +263,47 @@ TEST_F(OAuth2ClientTest, RequestAccessTokenInvalidResponse) {
       [&](auto* callback) { callback->onSuccess(request, std::move(mock_response)); }));
 }
 
+TEST_F(OAuth2ClientTest, RequestAccessTokenNetworkError) {
```

Comment:
I moved this up from below so it's grouped with the other access token tests instead of the refresh token tests.